spec/filters/grok_spec.rb

```diff
@@ -702,4 +702,66 @@
     end
   end
 
+  describe  "grok with nil coerced value" do
+    config <<-CONFIG
+      filter {
+        grok {
+          match => { "message" => "test (N/A|%{BASE10NUM:duration:float}ms)" }
+        }
+      }
+    CONFIG
+
+    sample "test 28.4ms" do
+      insist { subject["duration"] } == 28.4
+      insist { subject["tags"] }.nil?
+    end
+
+    sample "test N/A" do
+      insist { insist { subject.to_hash }.include?("duration") }.fails
```

Comment:
One other note, the pattern of `insist { insist { } }.fails` is older and predates the `reject` method added later in insist's life.

Examples here use the old style: https://github.com/jordansissel/ruby-insist/blob/master/spec/insist/predicates_spec.rb#L48-L50

```
# Old style, assert something fails
insist { insist { foo }.empty? }.fails

# New style: rejecting success
reject { foo }.empty?
```